pkg/kernel/networkservice/connectioncontextkernel/ipcontext/ipaddress/common.go

```diff
@@ -131,7 +139,10 @@ func waitForIPNets(ctx context.Context, ch chan netlink.AddrUpdate, l netlink.Li
 		select {
 		case <-ctx.Done():
 			return errors.Wrapf(ctx.Err(), "timeout waiting for update to add ip addresses %s to %s (type: %s)", ipNets, l.Attrs().Name, l.Type())
-		case update := <-ch:
+		case update, ok := <-ch:
```

Comment:
```suggestion
		case <-done: //TODO
```

Reply:
It wouldn't work - we still have `netlink.addrSubscribeAt` blocked on writing to `ch`.